sdk/servicebus/service-bus/samples/typescript/src/advanced/sessionManagerSample.ts

```diff
@@ -0,0 +1,155 @@
+import { ServiceBusClient, delay, ReceivedMessageWithLock } from "@azure/service-bus";
+import * as dotenv from "dotenv";
+import { env } from "process";
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+
+dotenv.config();
+
+const serviceBusConnectionString =
+  env["SERVICEBUS_CONNECTION_STRING"] || "<service bus connection string not in environment>";
+const queueName = env["QUEUE_NAME_WITH_SESSIONS"] || "<queue name not in environment>";
+const maxSessionsToProcessSimultaneously = 8;
+const sessionIdleTimeoutMs = 3 * 1000;
+const delayWhenNoSessionsAvailableMs = 5 * 1000;
+
+interface SessionMessageHandlers {
+  processInitialize(sessionId: string);
+  processMessage(msg: ReceivedMessageWithLock);
+  processError(err: Error, sessionId: string);
+  processClose(reason: "error" | "idle_timeout", sessionId: string);
+}
+
+function _createIdleTimer(
+  timeoutMs: number
+): { refresh(): void; expirationPromise: Promise<void> } {
+  let resolveExpirationPromise;
+
+  const expirationPromise = new Promise<void>((res, _rej) => {
+    resolveExpirationPromise = res;
+  });
+
+  let timer = setTimeout(resolveExpirationPromise, timeoutMs);
+
+  return {
+    refresh() {
+      clearTimeout(timer);
+      timer = setTimeout(resolveExpirationPromise, timeoutMs);
+    },
+    expirationPromise
+  };
+}
+
+async function _noSessionsAvailable(err: Error) {
```

Comment:
I'm not sure the leading underscore is warranted here

Reply:
I just wanted some way to say "this is a private detail".

I ended up eliminate this particular function but I'm guessing you want it gone from the others as well? Is there a better convention?

Reply:
Removed for now. :)